src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,103 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int GROWTH_COEFFICIENT = 2;
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int DEFAULT_SIZE = 0;
+
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        checkSize();
+        addNode(key, value);
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K, V> node = table[index];
+
+        while (node != null) {
+            if (Objects.equals(node.key, key)) {
+                return node.value;
+            }
+            node = node.next;
+        }
+
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void addNode(K key, V value) {
+        Node<K, V> newNode = new Node<>(key, value, null);
+        int index = getIndex(key);
+        Node<K, V> node = table[index];
+
+        if (node == null) {
+            table[index] = newNode;
+        } else {
+            while (true) {
+                if (Objects.equals(node.key, key)) {
+                    node.value = value;
+                    return;
+                } else if (node.next == null) {
+                    node.next = newNode;
+                    break;
+                }
+                node = node.next;
+            }
+        }
+
+        size++;
+    }
+
+    private void checkSize() {
+        if (size == (LOAD_FACTOR * table.length)) {
```

Comment:
just to be sure, because later on we may for example want to put data in chunks and not 1 by 1, and this check wouldn't work then, and generally I'd choose a more wide condition, if I have an opportunity
```suggestion
        if (size >= (LOAD_FACTOR * table.length)) {
```